src/commands/NewCamelProjectCommand.ts

```diff
@@ -16,29 +16,42 @@
  */
 'use strict';
 
-import { Uri, WorkspaceFolder, window, workspace } from "vscode";
-import { CamelExportJBangTask } from "../tasks/CamelExportJBangTask";
 import * as path from "path";
+import { TaskScope, Uri, commands, window, workspace } from "vscode";
+import { CamelExportJBangTask } from "../tasks/CamelExportJBangTask";
 
 export abstract class NewCamelProjectCommand {
 
 	async create() {
 		const input = await this.askForGAV();
 		if (input) {
-			let workspaceFolder: WorkspaceFolder | undefined;
-			if (workspace.workspaceFolders) {
-				// default to root workspace folder
-				workspaceFolder = workspace.workspaceFolders[0];
+
+			// Uses the user selected folder otherwise default to the root workspace folder
+			const outputFolder = await this.showDialogToPickFolder();
+			if (!outputFolder) {
+				await window.showErrorMessage('Operation canceled or invalid folder selection');
+				return;
 			}
-			const runtime = this.getRuntime();
-			if(workspaceFolder){
-				await new CamelExportJBangTask(workspaceFolder, input, runtime).execute();
+
+			const userChoice = await this.confirmDestructiveActionInSelectedFolder();
+
+			if (userChoice === undefined){
+				await window.showErrorMessage('Operation canceled');
```

Comment:
this message appears in case the user clicked on `Cancel`, I do not think that it is should be shown as an error. It might be an info message but I'm not sure that it is even needed as it is a potential action chosen from the user and not an internal exception or an illegal state.